File: app_dagger2_sample_01/src/main/java/com/rsh_engineering/tkachenkoni/app_dagger2_sample_01/MainActivity.java
package com.rsh_engineering.tkachenkoni.app_dagger2_sample_01;

import android.content.Context;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import javax.inject.Inject;

/**
 * Created by Nikolay Tkachenko on 02, October, 2020
 */
public class MainActivity extends AppCompatActivity {

    @Inject
    Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((App02) getApplication()).getComponent().inject(this);



    }
}

File: app_dagger2_sample_06_edu_kotlin/src/main/java/com/rsh_engineering/tkachenkoni/app_dagger2_sample_06_edu_kotlin/MainActivity.kt
package com.rsh_engineering.tkachenkoni.app_dagger2_sample_06_edu_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.rsh_engineering.tkachenkoni.app_dagger2_sample_06_edu_kotlin.di.DaggerCarComponent

import com.rsh_engineering.tkachenkoni.app_dagger2_sample_06_edu_kotlin.model.Car
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var car: Car


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var carComponent = DaggerCarComponent.create()
        carComponent.inject(this)

        car.start()

    }
}
